docs: Admin Partitions K8s requirement update (#12173)
hashicorp/consul: website/content/docs/enterprise/admin-partitions.mdx

One of the primary use cases for admin partitions is for enabling a service mesh across multiple Kubernetes clusters. The following requirements must be met to create admin partitions on Kubernetes:

> * If deploying both Consul servers on Kubernetes, Consul servers must be deployed wihtin a single Kubernetes cluster. Consul servers may be deployed external to Kubernetes and configured using the `externalServers` stanza. 

Reviewer: ```suggestion
* If deploying both Consul servers on Kubernetes, each Consul server must be deployed in its own Kubernetes cluster. Consul servers may be deployed external to Kubernetes and configured using the `externalServers` stanza. 
```
Thoughts on this? I tried to make it a bit more clearer.
